engineio/src/transports/websocket.rs

```diff
@@ -1,122 +1,69 @@
-use crate::error::{Error, Result};
-use crate::packet::Packet;
-use crate::packet::PacketId;
-use crate::transport::Transport;
-use bytes::{BufMut, Bytes, BytesMut};
-use std::borrow::Cow;
-use std::str::from_utf8;
-use std::sync::{Arc, Mutex, RwLock};
-use websocket::{
-    client::Url, dataframe::Opcode, header::Headers, receiver::Reader, sync::stream::TcpStream,
-    sync::Writer, ws::dataframe::DataFrame, ClientBuilder as WsClientBuilder, Message,
+use crate::{
+    async_transports::{transport::AsyncTransport, websocket::AsyncWebsocketTransport},
+    error::Result,
+    header::HeaderMap,
+    transport::Transport,
 };
+use bytes::Bytes;
+use std::sync::Arc;
+use tokio::runtime::Runtime;
+use url::Url;
 
 #[derive(Clone)]
 pub struct WebsocketTransport {
-    sender: Arc<Mutex<Writer<TcpStream>>>,
-    receiver: Arc<Mutex<Reader<TcpStream>>>,
-    base_url: Arc<RwLock<url::Url>>,
+    runtime: Arc<Runtime>,
+    inner: Arc<AsyncWebsocketTransport>,
 }
 
 impl WebsocketTransport {
     /// Creates an instance of `WebsocketTransport`.
-    pub fn new(base_url: Url, headers: Option<Headers>) -> Result<Self> {
+    pub fn new(base_url: Url, headers: Option<HeaderMap>) -> Result<Self> {
+        let runtime = tokio::runtime::Builder::new_current_thread()
```

Comment:
nit: do we want it only on one thread on multi threaded?

Reply:
True, I see no point in not running this multi-threaded..